apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveDao.kt

```diff
@@ -138,25 +142,56 @@ class OppgaveDaoImpl(private val connection: () -> Connection) : OppgaveDao {
         }
     }
 
-    override fun hentOppgaver(oppgaveTypeTyper: List<OppgaveType>): List<OppgaveIntern> {
+    override fun hentOppgaver(
+        oppgaveTypeTyper: List<OppgaveType>,
+        enheter: List<String>,
+        erSuperBruker: Boolean,
+    ): List<OppgaveIntern> {
         if (oppgaveTypeTyper.isEmpty()) return emptyList()
 
         with(connection()) {
             val statement =
                 prepareStatement(
-                    """
-                    SELECT o.id, o.status, o.enhet, o.sak_id, o.type, o.saksbehandler, o.referanse, o.merknad, o.opprettet, o.saktype, o.fnr, o.frist, o.kilde
-                    FROM oppgave o INNER JOIN sak s ON o.sak_id = s.id
-                    WHERE o.type = ANY(?)
-                    AND (
-                        s.adressebeskyttelse is null OR 
-                        (s.adressebeskyttelse is NOT NULL AND (s.adressebeskyttelse != ? AND s.adressebeskyttelse != ?))
-                    )
-                    """.trimIndent(),
+                    when (erSuperBruker) {
+                        true -> {
+                            """
+                            SELECT o.id, o.status, o.enhet, o.sak_id, o.type, o.saksbehandler, o.referanse, o.merknad, o.opprettet, o.saktype, o.fnr, o.frist, o.kilde
+                            FROM oppgave o INNER JOIN sak s ON o.sak_id = s.id
+                            WHERE o.type = ANY(?)
+                            AND (
+                                s.adressebeskyttelse is null OR 
+                                (s.adressebeskyttelse is NOT NULL AND (s.adressebeskyttelse != ? AND s.adressebeskyttelse != ?))
+                            )
+                            """.trimIndent()
+                        }
+                        false -> {
+                            """
+                            SELECT o.id, o.status, o.enhet, o.sak_id, o.type, o.saksbehandler, o.referanse, o.merknad, o.opprettet, o.saktype, o.fnr, o.frist, o.kilde
+                            FROM oppgave o INNER JOIN sak s ON o.sak_id = s.id
+                            WHERE o.type = ANY(?)
+                            AND o.enhet = ANY(?)
```

Comment:
kunne kanskje ha unngått å lage 2 spørringer ved å ha typ
`AND (? = true || o.enhet = ANY(?))` hvor første da er superbruker eller ei.
men gjør ikke meg noe denne approachen

Reply:
Enig i det, kommentaren poppet ikke opp før jeg rakk å trykke merge men her https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/3503